fetch.bs

```diff
@@ -2298,55 +2331,103 @@ steps:
    <a>connection</a>.
   </ol>
 
- <li><p>Let <var>connection</var> be null.
+ <li><p>Let <var>timingInfo</var> be a new <a for=/>connection timing info</a>.
 
- <li>
-  <p>Run these steps, but <a>abort when</a> the ongoing fetch is <a for=fetch>terminated</a>:
-
-  <ol>
-   <li>
-    <p>Set <var>connection</var> to a new <a for=/>connection</a> whose <a for=connection>key</a> is
-    <var>key</var>, <a for=connection>origin</a> is <var>origin</var>, and
-    <a for=connection>credentials</a> is <var>credentials</var>.
-    <a for=/>Record connection timing info</a> given <var>connection</var> and use
-    <var>connection</var> to establish an HTTP connection to <var>origin</var>.
-    [[!HTTP]] [[!HTTP-SEMANTICS]] [[!HTTP-COND]] [[!HTTP-CACHING]] [[!HTTP-AUTH]] [[!TLS]]
+ <li><p>Set <var>timingInfo</var>'s <a for="connection timing info">domain lookup start time</a> to
+ the <a for=/>unsafe shared current time</a>.
 
-    <p>If <var>http3Only</var> is true, then establish an HTTP/3 connection. [[!HTTP3]]
+ <li><p>Let <var>hosts</var> be the result of <a lt="resolve an origin">resolving</a>
+ <var>origin</var>.
 
-    <p>When establishing an HTTP/3 connection, include SETTINGS_ENABLE_WEBTRANSPORT with a value of
-    1 and H3_DATAGRAM with a value of 1 in the initial SETTINGS frame. [[!WEBTRANSPORT-HTTP3]]
-    [[!HTTP3-DATAGRAM]]
+ <li><p>If <var>hosts</var> is failure, then return failure.
 
-    <p>If <var>credentials</var> is false, then do <em>not</em> send a TLS client certificate.
-
-    <p>If establishing a connection does not succeed (e.g., a DNS, TCP, or TLS error), then return
-    failure.
-  </ol>
+ <li><p>Set <var>timingInfo</var>'s <a for="connection timing info">domain lookup end time</a> to
+ the <a for=/>unsafe shared current time</a>.
 
  <li>
-  <p><a>If aborted</a>, then:
+  <p>Let <var>connection</var> be the result of running this step: run <a>create a connection</a>
```

Comment:
This is probably a bad time to mention that Chromium starts racing a second connection if the first one takes more than 250ms. Maybe this can be treated as a kind of preconnect for standardisation purposes?

Reply:
I guess to allow that kind of behavior I shouldn't put an upper limit on the number of times create a connection can be invoked. I think that's reasonable.